test/e2e/upgrades/upgrade.go

```diff
@@ -57,3 +60,24 @@ type Test interface {
 	// aren't already cleaned up by the framework.
 	Teardown(f *framework.Framework)
 }
+
+// Skippable is an interface that an upgrade test can implement to be
+// able to indicate that it should be skipped.
+type Skippable interface {
+	// Skip should return true if test should be skipped. upgCtx
+	// provides information about the upgrade that is going to
+	// occur.
+	Skip(upgCtx UpgradeContext) bool
+}
+
+// UpgradeContext contains information about all the stages of the
+// upgrade that is going to occur.
+type UpgradeContext struct {
+	Versions []VersionContext
```

Comment:
Will this ever have more than two entries? Will it know the difference between a node upgrade and master upgrades?